hazelcast-client-new/src/main/java/com/hazelcast/client/cache/impl/AbstractClientInternalCacheProxy.java

```diff
@@ -409,11 +556,35 @@ protected ClientInvocationFuture invoke(ClientMessage req, Data keyData, int com
         } catch (Exception e) {
             throw ExceptionUtil.rethrow(e);
         }
-        return new ClientDelegatingFuture<Boolean>(future, clientContext.getSerializationService(),
-                putIfAbsentResponseDecoder);
+        ClientDelegatingFuture delegatingFuture =
+                new ClientDelegatingFuture<Boolean>(future, clientContext.getSerializationService(),
+                                                    putIfAbsentResponseDecoder);
+        if (async && statisticsEnabled) {
+            delegatingFuture.andThen(new ExecutionCallback<Object>() {
```

Comment:
I don't mean there are different executors, I say that this statistics are just simple calculation and no need to handle them by creating an extra object (ExecutionCallback) and processing it on another executor thread.